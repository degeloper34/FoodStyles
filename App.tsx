import {ApolloClient, ApolloProvider, InMemoryCache} from "@apollo/client";
import {StatusBar} from "expo-status-bar";
import {SafeAreaProvider} from "react-native-safe-area-context";
import useCachedResources from "./src/hooks/useCachedResources";
import Navigation from "./src/navigation";

const token =
  "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NTI3LCJ0eXBlIjoiQUNDRVNTIiwiaWF0IjoxNjU3MTQ0MzAyLCJleHAiOjE2NTc3NDkxMDJ9.XxvR4ah2U6NeUFAxKtI-sSzByFfbHGcz06LGpp1FMPVCar6mQyfctQgtKOkg2w5Wv_luULa0SgcSovS6VNUB9_BBlFdH7PYBjQO3N1N0Dwi53FBzA0rdnSkRGdj5Oe7hQ-9fshljlo5cJUYBbz4N0Ndpt323V5Eu8lGqJZ8Z5husoCiK93fZ0l3-Vr0M_sLvulZAELXGRG1QmYCwMwrp43HF9zyOLjcSHPVCvbpettqGxGv8Gne7O-XbykNv8ehG81vPcJDMSImwXGY8Yd47EoIrxdoN-3IlJumBK1Ql29hy5N7HBbk2oil3KMwBbw4MowB7rdlaT1i5_5Xr6q0iIg";

export const client = new ApolloClient({
  uri: "https://api-dev.foodstyles.com/graphql",
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer ${token}`,
  },
});

export default function App() {
  const isLoadingComplete = useCachedResources();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      <ApolloProvider client={client}>
        <SafeAreaProvider>
          <StatusBar style={"light"} />
          <Navigation />
        </SafeAreaProvider>
      </ApolloProvider>
    );
  }
}
