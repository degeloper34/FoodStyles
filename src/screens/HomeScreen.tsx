import {useQuery} from "@apollo/client";
import {LinearGradient} from "expo-linear-gradient";
import {StyleSheet, View, Image, Pressable, FlatList} from "react-native";
import {Card, RootStackScreenProps} from "../../types";
import {GET_CARDS_QUERY} from "../graphql/queries";
import {CustomText, Loading} from "../components/atoms";
import {Colors, Spacing, TextStyles} from "../styles";

export default function HomeScreen({
  navigation,
}: RootStackScreenProps<"HomeScreen">) {
  const onPressNewFoodStyle = () => {
    navigation.navigate("ModalScreen");
  };

  const {data, loading} = useQuery(GET_CARDS_QUERY);

  const {
    container,
    viewCard,
    viewCardName,
    linearGradient,
    viewBody,
    flatList,
    viewBottom,
    btnAdd,
  } = styles;

  const renderItem = ({item}: {item: Card}) => {
    return (
      <View style={viewCard}>
        <View style={viewCardName}>
          <CustomText
            text={item?.name}
            type={"bold"}
            style={TextStyles.TEXT_STYLE_6}
            numberOfLines={2}
          />
        </View>
        <Pressable onPress={() => navigation.navigate("ActionsModal", {item})}>
          <Image source={require("../assets/images/options.png")} />
        </Pressable>
      </View>
    );
  };

  return (
    <View style={container}>
      {loading && <Loading />}
      <LinearGradient
        colors={["rgb(250,119,69)", "rgb(243, 196, 66)", "rgb(248, 248, 248)"]}
        style={linearGradient}
      />

      <View style={viewBody}>
        <FlatList
          data={data?.cards}
          renderItem={renderItem}
          keyExtractor={(_, index) => String(index)}
          contentContainerStyle={flatList}
          showsVerticalScrollIndicator={false}
          initialNumToRender={10}
        />
      </View>

      <View style={viewBottom}>
        <Pressable style={btnAdd} onPress={onPressNewFoodStyle}>
          <Image source={require("../assets/images/add.png")} />
          <CustomText
            text={"New Food Style"}
            type={"bold"}
            style={TextStyles.TEXT_STYLE_7}
          />
        </Pressable>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  viewCard: {
    borderRadius: 6,
    backgroundColor: Colors.WHITE,
    shadowColor: Colors.GREYISH_40,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 7,
    shadowOpacity: 1,
    alignItems: "center",
    flexDirection: "row",
    padding: Spacing.s,
    marginBottom: Spacing.xs,
  },
  viewCardName: {
    flex: 1,
  },
  linearGradient: {
    flex: 2,
  },
  viewBody: {
    flex: 9,
    backgroundColor: "rgb(248, 248, 248)",
  },
  flatList: {
    paddingHorizontal: Spacing.l,
    paddingBottom: Spacing.l,
  },
  viewBottom: {
    flex: 1,
    backgroundColor: "white",
    shadowColor: "rgba(0, 0, 0, 0.1)",
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowRadius: 15,
    shadowOpacity: 1,
  },
  btnAdd: {
    borderRadius: 6,
    backgroundColor: Colors.WHITE,
    shadowColor: Colors.GREYISH_40,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 7,
    shadowOpacity: 1,
    marginHorizontal: Spacing.l,
    padding: Spacing.xs,
    position: "absolute",
    left: 0,
    right: 0,
    bottom: 40,
    flexDirection: "row",
    alignItems: "center",
  },
});
