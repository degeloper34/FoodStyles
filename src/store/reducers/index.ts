import {combineReducers} from "redux";

import memberReducers from "./memberReducers";

export const rootReducer = combineReducers({
  memberReducers,
});
