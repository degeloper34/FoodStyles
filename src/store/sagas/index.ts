import {all} from "redux-saga/effects";

import {memberWatcher} from "./memberSagas";

export default function* rootSaga() {
  yield all([memberWatcher()]);
}
