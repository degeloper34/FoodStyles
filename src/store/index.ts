export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

import {legacy_createStore as createStore} from "redux";
import {rootReducer} from "./reducers";

export const store = createStore(rootReducer);
