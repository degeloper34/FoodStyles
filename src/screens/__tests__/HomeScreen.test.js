import renderer from "react-test-renderer";
import React from "react";
import {Provider} from "react-redux";
import {store} from "../../store";
import HomeScreen from "../HomeScreen";
import {ApolloProvider} from "@apollo/client";
import {apolloClient} from "../../../App";

describe("HomeScreen Tests", () => {
  test("HomeScreen | Render without issues", () => {
    const setOptions = jest.fn();

    const screen = renderer
      .create(
        <ApolloProvider client={apolloClient}>
          <Provider store={store}>
            <HomeScreen
              navigation={{
                setOptions,
              }}
            />
          </Provider>
        </ApolloProvider>
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
