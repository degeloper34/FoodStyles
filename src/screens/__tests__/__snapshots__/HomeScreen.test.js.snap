// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Tests HomeScreen | Render without issues 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    color="#ffffff"
    size="large"
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
  />
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4294604613,
          4294165570,
          4294506744,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 2,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgb(248, 248, 248)",
        "flex": 9,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingBottom": 20,
          "paddingHorizontal": 20,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      initialNumToRender={10}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 15,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 6,
          "bottom": 40,
          "flexDirection": "row",
          "left": 0,
          "marginHorizontal": 20,
          "padding": 8,
          "position": "absolute",
          "right": 0,
          "shadowColor": "rgba(176, 176, 176, 0.4)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#434343",
              "fontSize": 18,
              "fontStyle": "normal",
              "fontWeight": "bold",
              "letterSpacing": 0,
            },
            Object {
              "fontFamily": "proxima-bold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        New Food Style
      </Text>
    </View>
  </View>
</View>
`;
